add friend-app files

webdevagent opened the pull request:
### friend-app-task

[Demo](https://webdevagent.github.io/friends-app/) |
[Code base](https://github.com/webdevagent/frontend-2019-homeworks/blob/friends-app-task/submissions/webdevagent/friend-app-task/index.js)

The code is submitted in a dedicated feature branch.
Please, review.

webdevagent commented:
So Here I made basic friend app, also in this app, you have the possibility to send a friend request, just push the connect button and you will your future friends in request tab. There is a lot of  repeated blocks of code, sorry for that, Promise that I'll fix everything))

IgorKurkov commented:
Great idea and some unify styles :) Great :) Sorry for long delay, but now I see some troubles with layout, take a look please 
![image](https://user-images.githubusercontent.com/29090227/51283087-aead9380-19ef-11e9-8ebc-e236ecc1e22c.png)
![image](https://user-images.githubusercontent.com/29090227/51283097-bff6a000-19ef-11e9-81b9-e21c480be748.png)
Tomorrow you will be first :)

webdevagent commented:
Hello @IgorKurkov, thank's for layout bug finding 😄.  I have been remove overflow: scrolling, so there is no more double scroll line. Also, I fixed the problem with first- time hide button pushing(container with friends was jumping higher, and right now it fixed). And I added a little bit more mobile adaptivity)

IgorKurkov commented on submissions/webdevagent/friend-app-task/index.js:
```diff
@@ -0,0 +1,145 @@
+    fillUsers(data.results);
+    resetArray = Users.slice();
+  });
+friendsContainer.addEventListener('click', flipCard);
```
set listeners below all the code at the bottom side

IgorKurkov commented on submissions/webdevagent/friend-app-task/index.js:
```diff
@@ -0,0 +1,145 @@
+friendsContainer.addEventListener('click', flipCard);
+
+navigation.addEventListener('click', ({target}) => {
+  if (target.className == 'a-z' || target.className == 'z-a') sortListDir(target);
```
move callback foo to separated foo and make it simpler, try to split it to smaller chunks

IgorKurkov commented on submissions/webdevagent/friend-app-task/index.js:
```diff
@@ -0,0 +1,145 @@
+  }
+  if (target.className == 'reset') renderNewFlist(resetArray);
+  if (target.className == 'hide') {
+    openButton.classList.remove('remove-card');
```
move to separated named foo like - `hideCard()`

IgorKurkov commented on submissions/webdevagent/friend-app-task/index.js:
```diff
@@ -0,0 +1,145 @@
+  }
+});
+
+navigation.addEventListener('keyup', inputSearch);
```
you have two event listeners on one element - think about how t\you can make one

IgorKurkov commented on submissions/webdevagent/friend-app-task/index.js:
```diff
@@ -0,0 +1,145 @@
+});
+
+navBar.addEventListener('click', ({target}) => {
+  function classChanger(friends) {
```
you define a function inside a callback - move it to the upper level fo ability to reuse it

IgorKurkov commented on submissions/webdevagent/friend-app-task/index.js:
```diff
@@ -0,0 +1,145 @@
+    let names = friendsContainer.querySelectorAll('.name');
+    names.forEach(num => {
+      if (num.textContent.toUpperCase().indexOf(value) > -1) {
+        friendsContainer.querySelector(`.flip-box[data-order='${num.dataset.order}']`).classList.remove('remove-card');
```
I understand the point, finally. You search through DOM, make changes in DOM, make a lot of injects and CSS repaints. It's not clear work with DOM but its not your fault. Now, all the frameworks want to isolate action to DOM directly for preventing side-effects. For emulating this logic the main idea of this app was that you can use a tiny MVC structure. When you manipulate with data, push sorted/filtered data to the view in one action, and again receive an action from view - make changes in data and again redraw the view. for this, you have the model - the array of users, controller - functions for sorting/filtering/rendering cards and view - HTML container where you need to insert your rendered cards. I understand that you can't rewrite all the structure because you already go in CSS/DOM way. We can discuss this case 26.01 together. It's not a review comment - only some suggestions that you can think about it.

webdevagent commented:
Hello @IgorKurkov, no problem😄. Thank's for code reviewing. Your advices were very helpful, thanks of them my code become more structured and readable. I have updated my readme(add gifs and some descriptions of functionality), also I removed array initialization with 0 value, update some arrays, add catch error function to fetch. For some app data, I defined an object with a collection of useful properties that I use in my app.  I have been set all event listeners in the bottom and also remove repeating of event listening assignment to one element.  Removed code repeating functions and combined in one using DRY methodology.
